File: src-openacc/exercise/solution/saxpy_acc.c
#include <stdlib.h> 

int main(int argc, char **argv) 
{

  int N=1000;
  float a = 3.0f;
  float x[N], y[N];
  //float *x = (float*)malloc(N * sizeof(float)); 
  //float * restrict y = (float*)malloc(N * sizeof(float)); 

  for (int i = 0; i < N; ++i) { 
     x[i] = 2.0f; 
     y[i] = 1.0f; 
  } 

#pragma acc kernels 
//#pragma acc parallel for
  for (int i = 0; i < N; ++i) { 
     y[i] = a * x[i] + y[i];
  } 

  printf("%f \t %f \n", y[0], y[N-1]);

  return 0; 

}

File: src-openacc/exercise/solution/saxpy_acc.f90
program main

integer :: n=1000, i 
real :: a=3.0
real, allocatable :: x(:), y(:)

allocate(x(n),y(n))

!do i=1,n 
!  x(i) = 2.0
!  y(i) = 1.0
!enddo 

x(1:n)=2.0;
y(1:n)=1.0;

!-- !$acc kernels 
!-- !$acc parallel loop
!$acc parallel do
do i=1,n 
  y(i) = a*x(i)+y(i)
enddo 
!-- !$acc end kernels
!-- !$acc end parallel loop
!$acc end parallel do

print*, y(1), y(n)

end program main
